### ISSUE TITLE
IntelliJ IDEA 2016.1: Opening settings dialog throws java.lang.AssertionError: Cannot find configurable: org.codinjutsu.tools.jenkins.JenkinsComponent

### ISSUE THREAD

**Issue by smheaton** [MCMicS/jenkins-control-plugin#116]
Steps to Reproduce:
1.  In IntelliJ IDEA 2016.1, expand the Jenkins window.
2.  Click on the "Jenkins Settings" button.

Result:

Cannot find configurable: org.codinjutsu.tools.jenkins.JenkinsComponent
java.lang.AssertionError: Cannot find configurable: org.codinjutsu.tools.jenkins.JenkinsComponent
    at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:112)
    at org.codinjutsu.tools.jenkins.view.action.OpenPluginSettingsAction.showSettingsFor(OpenPluginSettingsAction.java:45)
    at org.codinjutsu.tools.jenkins.view.action.OpenPluginSettingsAction.actionPerformed(OpenPluginSettingsAction.java:41)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:191)
    at com.intellij.openapi.actionSystem.impl.ActionButton.a(ActionButton.java:162)
    at com.intellij.openapi.actionSystem.impl.ActionButton.a(ActionButton.java:126)
    at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:300)
    at java.awt.Component.processEvent(Component.java:6300)
    at java.awt.Container.processEvent(Container.java:2236)
    at java.awt.Component.dispatchEventImpl(Component.java:4891)
    at java.awt.Container.dispatchEventImpl(Container.java:2294)
    at java.awt.Component.dispatchEvent(Component.java:4713)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
    at java.awt.Container.dispatchEventImpl(Container.java:2280)
    at java.awt.Window.dispatchEventImpl(Window.java:2750)
    at java.awt.Component.dispatchEvent(Component.java:4713)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue$4.run(EventQueue.java:731)
    at java.awt.EventQueue$4.run(EventQueue.java:729)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
    at com.intellij.ide.IdeEventQueue.a(IdeEventQueue.java:857)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:654)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by christopherfowers** [comment]
Is there an update on this?

**Comment by OrangeDog** [comment]
Worth noting that the plugin is completely non-functional, not just the settings pane.

**Comment by christophpickl** [comment]
same here, same exception, same intellij version, ... when trying to open the settings it just fails. plugin not usable at all :'-(

**Comment by christopherfowers** [comment]
You can manually create the config xml for this plugin so that it will work. This is by no means a good solution. But the plugin is not completely useless. I will post back with a tutorial on this temporary workaround.

**Comment by christopherfowers** [comment]
This issue is the same as it was for this plugin 2 years ago. See [this](https://github.com/dboissier/jenkins-control-plugin/issues/88). Reported by smheaton there as it is here. The first response (from Jbo31) is a working temporary solution to the problem. It is what I am using now. You will still not have access to the settings menu. This is therefore not a long term solution.

**Comment by OrangeDog** [comment]
What @christopherfowers didn't mention is that you need to update the project version from 4 to 5.

**Comment by christophpickl** [comment]
great, works like a charm :)
one of the first lesson learned out there in the real life business world: "_a working workaround is just as good as a proper solution_" -i still dont agree, but anyway...

just as you mentioned: create a file `<YOUR_PROJECT>/.idea/jenkinsSettings.xml` and add the following content:

```
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="Jenkins.Application.Settings">
    <option name="serverUrl" value="http://<HOST>:<IP>" />
    <option name="jobRefreshPeriod" value="2" />
    <option name="rssRefreshPeriod" value="2" />
    <option name="rssSettings">
    <RssSettings>
      <option name="displaySuccessOrStable" value="false" />
    </RssSettings>
    </option>
  </component>
</project>
```
and the plugin will work as expected. all configuration has to be done via this config file directly, as the UI still wont work after this hack.

much appreciated,
christoph

**Comment by christopherfowers** [comment]
Going to attempt to fix the issues with the plugin later today.

**Comment by JProof** [comment]
@christophpickl how do you resolve the jenkins-login credentials? with your config file it does not throws errors but if i try to conifgure the plugin int throws 
 AssertionError: Cannot find configurable: org.codinjutsu.tools.jenkins.JenkinsComponent

**Comment by christophpickl** [comment]
@JProof i just dont need any login credentials for this jenkins instance ;) viewing and building is allowed for anonymous as the server is just reachable within the LAN. but i guess adding something like ```<option name="username/password value="xxx" />``` might solve this issue...

**Comment by amirnadivtradair** [comment]
Try using Community edition , It seem to work there

**Comment by ihorpts** [comment]
+1, please fix it for IntelliJ 2016

**Comment by amirnadivtradair** [comment]
yep , version 15 community edition works GR8 :-)

**Comment by JProof** [comment]
i know that it works in earlier versions. thx

**Comment by stingus** [comment]
`<option name="username" value="xxx" />` and `<option name="password" value="xxx" />` doesn't seem to work.

**Comment by SlavaRa** [comment]
please fix it for IntelliJ 2016.1

**Comment by thebenwaters** [comment]
+1

**Comment by oliren** [comment]
Not work with PyCharm Community Edition 2016.1.4
Build #PC-145.1504, built on May 25, 2016
JRE: 1.8.0_76-release-b198 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

**Comment by dboissier** [comment]
Hi all,

Deeply sorry for not being active these months. SNAPSHOT build for 2016 product is [available](https://github.com/dboissier/jenkins-control-plugin/raw/master/snapshot/jenkins-control-plugin-0.9.5-SNAPSHOT-distribution-idea2016.zip).

Please tell me if it works on your IDE.

Thanks.

**Comment by davidvanlaatum** [comment]
Can now get past configuring but when I hit refresh

null
java.lang.NullPointerException
	at org.codinjutsu.tools.jenkins.logic.JenkinsJsonParser.getParameters(JenkinsJsonParser.java:203)
	at org.codinjutsu.tools.jenkins.logic.JenkinsJsonParser.getJob(JenkinsJsonParser.java:185)
	at org.codinjutsu.tools.jenkins.logic.JenkinsJsonParser.createViewJobs(JenkinsJsonParser.java:283)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.loadJenkinsView(RequestManager.java:124)
	at org.codinjutsu.tools.jenkins.logic.RequestManager.loadJenkinsView(RequestManager.java:239)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.loadJobs(BrowserPanel.java:457)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.access$700(BrowserPanel.java:65)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$LoadSelectedViewJob.run(BrowserPanel.java:565)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:563)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:126)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


Jenkins 2.7

**Comment by stingus** [comment]
Works on  PHPStorm 2016.1 with Jenkins 2.5. Thanks!

**Comment by dboissier** [comment]
Hi,

it seems the Jenkins API has been slightly changed since version 2. I will have a look on it.

**Comment by oliren** [comment]
I can't establish connection to my Jenkins server. I see "Missing or bad
crumb data" when I click to Test connection button.

2016-06-03 11:48 GMT+03:00 David Boissier <notifications@github.com>:

**Comment by dboissier** [comment]
@oliren This is when security is enabled on your Jenkins server.

BTW, which version of Jenkins you use?

**Comment by thebenwaters** [comment]
WebStorm 2016.1.3
Build #WS-145.1616, built on May 27, 2016
JRE: 1.8.0_76-release-b198 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

Still dones't work same error

```
12:13:13 AssertionError: Cannot find configurable: org.codinjutsu.tools.jenkins.JenkinsComponent
```

**Comment by dboissier** [comment]
@thebenwaters Di you try the lastest snapshot build available [here](https://github.com/dboissier/jenkins-control-plugin)?

**Comment by thebenwaters** [comment]
@dboissier Snapshot seems to be working. Haven't hardcore checked it out, but I was at least able to configure it.

**Comment by dboissier** [comment]
@davidvanlaatum Hi, new snapshots are available on the home page. Please check whether the bug still occurs.
If it is ok, I will release today.

**Comment by davidvanlaatum** [comment]
Different exception this time
```
null
java.lang.NullPointerException
	at org.codinjutsu.tools.jenkins.model.Build.getStateIcon(Build.java:87)
	at org.codinjutsu.tools.jenkins.model.Job.getStateIcon(Job.java:75)
	at org.codinjutsu.tools.jenkins.view.JenkinsTreeRenderer.customizeCellRenderer(JenkinsTreeRenderer.java:64)
	at com.intellij.ui.ColoredTreeCellRenderer.getTreeCellRendererComponent(ColoredTreeCellRenderer.java:116)
	at javax.swing.plaf.basic.BasicTreeUI$NodeDimensionsHandler.getNodeDimensions(BasicTreeUI.java:2807)
	at javax.swing.tree.AbstractLayoutCache.getNodeDimensions(AbstractLayoutCache.java:492)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.updatePreferredSize(VariableHeightLayoutCache.java:1360)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1487)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1288)
	at javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:743)
	at javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:644)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:3917)
	at javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:580)
	at javax.swing.tree.DefaultTreeModel.nodeStructureChanged(DefaultTreeModel.java:365)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$4.run(BrowserPanel.java:203)
	at org.codinjutsu.tools.jenkins.util.GuiUtil.runInSwingThread(GuiUtil.java:50)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.setSortedByStatus(BrowserPanel.java:200)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.fillJobTree(BrowserPanel.java:497)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.access$900(BrowserPanel.java:65)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$LoadSelectedViewJob$1.run(BrowserPanel.java:581)
	at org.codinjutsu.tools.jenkins.util.GuiUtil.runInSwingThread(GuiUtil.java:50)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$LoadSelectedViewJob.onSuccess(BrowserPanel.java:578)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1$2.run(ProgressManagerImpl.java:153)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:345)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:329)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)



java.lang.NullPointerException
	at org.codinjutsu.tools.jenkins.model.Build.getStateIcon(Build.java:87)
	at org.codinjutsu.tools.jenkins.model.Job.getStateIcon(Job.java:75)
	at org.codinjutsu.tools.jenkins.view.JenkinsTreeRenderer.customizeCellRenderer(JenkinsTreeRenderer.java:64)
	at com.intellij.ui.ColoredTreeCellRenderer.getTreeCellRendererComponent(ColoredTreeCellRenderer.java:116)
	at javax.swing.plaf.basic.BasicTreeUI$NodeDimensionsHandler.getNodeDimensions(BasicTreeUI.java:2807)
	at javax.swing.tree.AbstractLayoutCache.getNodeDimensions(AbstractLayoutCache.java:492)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.updatePreferredSize(VariableHeightLayoutCache.java:1360)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1487)
	at javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1288)
	at javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:743)
	at javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:644)
	at javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:3917)
	at javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:580)
	at javax.swing.tree.DefaultTreeModel.nodeStructureChanged(DefaultTreeModel.java:365)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$4.run(BrowserPanel.java:203)
	at org.codinjutsu.tools.jenkins.util.GuiUtil.runInSwingThread(GuiUtil.java:50)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.setSortedByStatus(BrowserPanel.java:200)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.fillJobTree(BrowserPanel.java:497)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel.access$900(BrowserPanel.java:65)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$LoadSelectedViewJob$1.run(BrowserPanel.java:581)
	at org.codinjutsu.tools.jenkins.util.GuiUtil.runInSwingThread(GuiUtil.java:50)
	at org.codinjutsu.tools.jenkins.view.BrowserPanel$LoadSelectedViewJob.onSuccess(BrowserPanel.java:578)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1$2.run(ProgressManagerImpl.java:153)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.a(LaterInvocator.java:345)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:329)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:857)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:658)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:386)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by davidvanlaatum** [comment]
Not sure if this is it but views all but one view works the "All" view doesn't which contains folders

**Comment by dboissier** [comment]
@davidvanlaatum which version of Jenkins do you currently use?
If version 2 some API changes could break the current plugin

**Comment by davidvanlaatum** [comment]
2.7

**Comment by dboissier** [comment]
ok. I create a new issue labeled "Fix compatibility with Jenkins 2"

**Comment by yurybubnov** [comment]
I'm using Jenkins ver. 1.645 and see same error. Plugin version 0.9.7
2016-06-07 16:42:50,763 [ 530837]  ERROR - pplication.impl.LaterInvocator - IntelliJ IDEA 2016.1.3  Build #IU-145.1617.8
2016-06-07 16:42:50,763 [ 530837]  ERROR - pplication.impl.LaterInvocator - JDK: 1.8.0_76-release
2016-06-07 16:42:50,763 [ 530837]  ERROR - pplication.impl.LaterInvocator - VM: OpenJDK 64-Bit Server VM
2016-06-07 16:42:50,763 [ 530837]  ERROR - pplication.impl.LaterInvocator - Vendor: JetBrains s.r.o
2016-06-07 16:42:50,763 [ 530837]  ERROR - pplication.impl.LaterInvocator - OS: Mac OS X
2016-06-07 16:42:56,998 [ 537072]  ERROR - pplication.impl.LaterInvocator - null
java.lang.NullPointerException
	at org.codinjutsu.tools.jenkins.model.Build.getStateIcon(Build.java:87)
	at org.codinjutsu.tools.jenkins.model.Job.getStateIcon(Job.java:75)
	at org.codinjutsu.tools.jenkins.view.JenkinsTreeRenderer.customizeCellRenderer(JenkinsTreeRenderer.java:64)
	at com.intellij.ui.ColoredTreeCellRenderer.getTreeCellRendererComponent(ColoredTreeCellRenderer.java:116)
	at javax.swing.plaf.basic.BasicTreeUI$NodeDimensionsHandler.getNodeDimensions(BasicTreeUI.java:2807)

**Comment by dbeuve** [comment]
I have had today the same error. I created issue #122. I guess issue #122 is a duplicate.The intellij version I am using is

IntelliJ IDEA 2016.1.2
Build #IC-145.972, built on May 14, 2016
JRE: 1.8.0_76-release-b47 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

**Comment by dboissier** [comment]
@davidvanlaatum: could you please run the followings urls from your browser (once you are authenticated if needed) and send me the output results:
`http://<jenkins_url>/<Your_View>/api/json?tree=nodeName,nodeDescription,primaryView%5Bname,url%5D,views%5Bname,url,views%5Bname,url%5D%5D`

`http://<jenkins_url>/<job_name>/api/json?tree=name,displayName,url,color,buildable,inQueue,healthReport%5Bdescription,iconUrl%5D,lastBuild%5Bid,url,building,result,number%5D,property%5BparameterDefinitions%5Bname,type,defaultParameterValue%5Bvalue%5D,choices%5D%5D`

Remove confidential data if needed.

Thanks

**Comment by MCMicS** [comment]
close this because it seems fixed and 2016.1 is not supported anymore
If any error occurs (with current versions) please report a new issue
